### PR TITLE
Fix symfony security vulns

### PR DESCRIPTION
**Pull request by BjornTwachtmann**
This got a lot messier than I thought. Includes upgrade to Symfony 4.3

### CODE REVIEW

File: src/Controller/BaseController.php [hunk 1]
```diff
@@ -125,4 +126,14 @@ protected function cachedRedirectToRoute($route, array $parameters = [], $status
     {
         return $this->cachedRedirect($this->generateUrl($route, $parameters), $status);
     }
+
+    private function fixLocaleString(string $locale): string
+    {
+        $locale = str_replace('-', '_', $locale);
+        if (strpos($locale, '_') !== false) {
+            $locale = substr($locale, 0, -strlen(strstr($locale, '_')));
```
Comment (Tasiobg) [inline src/Controller/BaseController.php]: I had no idea you could do that to get a negative number, interesting

File: src/EventSubscriber/ResponseSubscriber.php [hunk 2]
```diff
@@ -2,14 +2,14 @@
 declare(strict_types = 1);
 namespace App\EventSubscriber;
 
-use App\Translate\TranslateProvider;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
 use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Translation\TranslatorInterface;
```
Comment (Tasiobg) [inline src/EventSubscriber/ResponseSubscriber.php]: This one seems deprecated, can you use the same you used in the base controller?
`Symfony\Contracts\Translation\TranslatorInterface;`

Comment (BjornTwachtmann) [reply src/EventSubscriber/ResponseSubscriber.php]: I did try that, it didn't work for some reason, I think maybe the interface didn't have all the needed methods on it

File: src/Service/CommentsService.php [hunk 3]
```diff
@@ -5,10 +5,10 @@
 
 use App\BlogsService\Domain\Blog;
 use App\BlogsService\Domain\Post;
-use App\Translate\TranslateProvider;
 use BBC\ProgrammesMorphLibrary\MorphClient;
 use GuzzleHttp\Promise\PromiseInterface;
 use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
```
Comment (Tasiobg) [inline src/Service/CommentsService.php]:
```suggestion

```

File: tests/Translate/TranslatableTraitTest.php [hunk 4]
```diff
@@ -4,17 +4,17 @@
 namespace Tests\App\Translate;
 
 use App\Translate\TranslatableTrait;
-use App\Translate\TranslateProvider;
 use PHPUnit\Framework\TestCase;
 use ReflectionClass;
-use RMP\Translate\Translate;
+use Symfony\Component\Translation\Translator;
```
Comment (Tasiobg) [inline tests/Translate/TranslatableTraitTest.php]:
```suggestion
```

File: composer.json [hunk 5]
```diff
@@ -34,22 +34,22 @@
         "ext-apcu": "*",
         "ext-json": "*",
         "bbc-rmp/cloudwatch-monitoringhandler": "^1.0",
-        "bbc-rmp/translate": "^1.8.9",
+	"symfony/translation": "4.3.2",
```
Comment (Tasiobg) [inline composer.json]:
```suggestion
	"symfony/translation": "4.3.2",
```
THIS IS UNACCEPTABLE!! not enough spaces :-P

**Pull request merged**